src/main/haskell/kore/src/Kore/Step/Simplification/Equals.hs

```diff
@@ -472,13 +514,13 @@ constructorAtTheTop
             (OrOfExpandedPattern level variable, SimplificationProof level)
     combineWithAnd tools' (thing1, _proof1) (thing2, _proof2) =
         And.simplifyEvaluated tools' thing1 thing2
-constructorAtTheTop _ _ _ = Nothing
+constructorAtTheTopAssumesNoBottom _ _ _ = Nothing
 
 
-firstMaybe :: [Maybe a] -> a -> a
-firstMaybe [] x = x
-firstMaybe (Just x : _) _ = x
-firstMaybe (_ : xs) x = firstMaybe xs x
+firstMaybe :: [Maybe a] -> Maybe a
```

Comment:
I think you can define `firstMaybe` as a composition of predefined functions: `listToMaybe . catMaybes`